plan/relations.go

```diff
@@ -117,6 +118,25 @@ func (b *baseReadRel) toReadRelProto() *proto.ReadRel {
 	return out
 }
 
+func (b *baseReadRel) GetInputs() []Rel {
+	return []Rel{}
+}
+
+func (b *baseReadRel) copyExpressions(rewriteFunc func(expr.Expression) (expr.Expression, error)) ([]expr.Expression, error) {
+	filter, err := rewriteFunc(b.filter)
+	if err != nil {
+		return nil, err
+	}
+	bestEffortFilter, err := rewriteFunc(b.bestEffortFilter)
+	if err != nil {
+		return nil, err
+	}
+	if filter == b.filter && bestEffortFilter == b.bestEffortFilter {
+		return nil, nil
+	}
+	return []expr.Expression{filter, bestEffortFilter}, nil
```

Comment:
could we shift this check up and just return `[]expr.Expression{filter, bestEffortFilter}` regardless (same as with the `CopyWithExpressionRewrite` method, just return the original ones if no changes are needed